repos/c2sm/packages/flexpart-cosmo/package.py

```diff
@@ -0,0 +1,43 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+#
+from spack import *
+from distutils.dir_util import copy_tree
+
+
+class FlexpartCosmo(MakefilePackage):
+    """flexpart is a Lagrangian dispersion model"""
+
+    homepage = 'https://github.com/C2SM-RCM/flexpart'
+    url = 'https://github.com/C2SM-RCM/flexpart/archive/refs/tags/V8C3-preop.tar.gz'
+    git = 'ssh://git@github.com/C2SM-RCM/flexpart.git'
+
+    version('main', branch='spack_compile')
+
+    depends_on('eccodes@2.25.0 +fortran')
```

Comment:
`depends_on('eccodes@2.25.0 +fortran')` means, that flexpart-cosmo needs eccodes 2.25.0 with the Fortran interface. I think the Fortran part is correct. But the version specifier implies that it can not be build with any other version of eccodes. I don't know if this is true or not, so I'm asking, can flexpart-cosmo also work with other version of eccodes? Because more often than not, software is built forward or backward compatible. If this is the case here, then maybe a less restrictive dependency would allow for greater flexibility.
Would `depends_on('eccodes +fortran')` also meet the requirements of flexpart-cosmo? Or maybe `depends_on('eccodes@2.19.0: +fortran')`, which is a minimal requirement of eccodes 2.19.0; or newer?
Or is `depends_on('eccodes@2.25.0: +fortran')` maybe an option, as flexpart-cosmo tries to be forward compatible to eccodes; and we may assume this is true until proven otherwise?

Reply:
@pirmink What do you think about other eccodes versions for flexpart-cosmo? What I could maybe do is to run our two forward and backward trajectory tests for different versions of eccodes. But so far testing only means that output is produced.

Reply:
I had a look at the repo and I think `depends_on('eccodes +fortran')` is the best choice for now, because the code at https://github.com/C2SM-RCM/flexpart does not have any special requirements on the version of eccodes. Thus I think it's fair to assume there really aren't any.